File: Kotlin/greet-name.kt
fun main() {
    val greeting = "Hello"
    var name = "John"
    println("$greeting, $name!") // Hello, John!

    // greeting = "Hi" // error: "val cannot be reassigned"
    name = "Karla"
    println("$greeting, $name!") // Hello, Karla!
}

File: Mercury/greet_name.m
:- module greet_name.

:- interface.
:- import_module io.
:- pred main(io::di, io::uo) is det.

:- implementation.
:- import_module string.

main(!IO) :-
    Name = "John",
    Greeting = "Hello, " ++ Name ++ "!",
    io.write_string(Greeting, !IO),
    io.write_string("\n", !IO).
